eman2/protocols/protocol_tomo_extraction.py

```diff
@@ -151,51 +157,89 @@ def readSetOfTomograms(self, workDir, tomogramsSet):
         self.outputSubTomogramsSet.setCoordinates3D(self.inputCoordinates)
-        self.readSetOfTomograms(self._getExtraPath(pwutils.join('sptboxer_01', 'basename.hdf')),
-                                self.outputSubTomogramsSet)
+
+        for item in self.inputSet:
+
+            for ind, tomoFile in enumerate(self.tomoFiles):
+                if tomoFile == item.getFileName():
+                    break
+
+            coordSet = self.lines[ind]
+
+            self.readSetOfTomograms(self._getExtraPath(pwutils.replaceBaseExt(tomoFile,"hdf")),
+                                    self.outputSubTomogramsSet, coordSet)
+
         self._defineOutputs(outputSetOfSubtomogram=self.outputSubTomogramsSet)
         self._defineSourceRelation(self.inputCoordinates, self.outputSubTomogramsSet)
 
 
     def writeSetOfCoordinates3D(self):
-        self.coordsFileName = self._getExtraPath(
-            pwutils.replaceBaseExt(self.getInputTomogram().getFileName(), 'coords'))
-        self.coordDict = []
-        out = file(self.coordsFileName, "w")
-        for coord3DSet in self.inputCoordinates.get().iterCoordinates():
-            out.write("%d\t%d\t%d\n" % (coord3DSet.getX(), coord3DSet.getY(), coord3DSet.getZ()))
-            self.coordDict.append(coord3DSet.clone())
-        out.close()
+
+        self.lines = []
+        self.tomoFiles = []
+        self.inputSet = self.getInputTomograms()
+
+        for item in self.inputSet:
+            coordDict = []
+            tomo = item.clone()
+            self.coordsFileName = self._getExtraPath(
+                pwutils.replaceBaseExt(tomo.getFileName(), 'coords'))
+
+            out = file(self.coordsFileName, "w")
+            for coord3DSet in self.inputCoordinates.get().iterCoordinates():
+                if tomo.getFileName() == coord3DSet.getVolName():
+                    out.write("%d\t%d\t%d\n" % (coord3DSet.getX(), coord3DSet.getY(), coord3DSet.getZ()))
+                    coordDict.append(coord3DSet.clone())
+
+            if coordDict:
+                self.lines.append(coordDict)
+                self.tomoFiles.append(tomo.getFileName())
+
+            out.close()
 
     # --------------------------- STEPS functions -----------------------------
     def extractParticles(self):
         TsCoord = self.inputCoordinates.get().getSamplingRate()
-        TsTomo = self.getInputTomogram().getSamplingRate()
-        self.cshrink = float(TsCoord/(TsTomo*self.downFactor.get()))
-        fnTomo = self.getInputTomogram().getFileName()
-        args = '%s --coords %s --boxsize %d' % (fnTomo, pwutils.replaceBaseExt(self.getInputTomogram().getFileName(), 'coords'),
-            self.boxSize.get())
-        if self.doInvert:
-            args += ' --invert'
 
-        if self.doNormalize:
-            args += ' --normproc %s' % self.getEnumText('normproc')
+        for item in self.inputSet:
+            args = ""
+            tomo = item.clone()
+            TsTomo = tomo.getSamplingRate()
```

Comment:
I guess you got this code from @EstrellaFG If you can change it as she did https://github.com/scipion-em/scipion-em-eman2/pull/33/commits/9c158b38a1db38591511411de61b3ca96e084186 it would improve readability